zip-0317b.rst

```diff
@@ -0,0 +1,161 @@
+::
+
+  ZIP: 0317b
+  Title: ZSA Extension Proportional Fee Mechanism
+  Owners: Daniel Benarroch <daniel@qed-it.com>
+          Pablo Kogan <pablo@qed-it.com>
+          Jonathan Rouach <jon@qed-it.com>
+  Credits: Daira Hopwood
+           Jack Grigg
+           Aditya Bharadwaj
+  Status: Draft
+  Category: Standards / Wallet
+  Created: 2022-11-25
+  License: MIT
+  Discussions-To: <>
+
+
+Terminology
+===========
+
+The key words "SHOULD", "SHOULD NOT", and "RECOMMENDED" in this document
+are to be interpreted as described in RFC 2119. [#RFC2119]_
+
+The term "conventional transaction fee" in this document is in reference
+to the value of a transaction fee that is conventionally used by wallets,
+and that a user can reasonably expect miners on the Zcash network to accept
+for including a transaction in a block.
+
+We define the following additional terms:
+
+- The term "network upgrade" in this document is to be interpreted as described in ZIP 200 [#zip-0200]_.
+- The term "Orchard" in this document is to be interpreted as described in ZIP 224 [#zip-0224]_.
+- The term "ZSA" in this document is to be interpreted as described in ZIP 226 [#zip-0226]_.
+- The term "Issuance" and "Issuance Action" in this document are to be interpreted as described in ZIP 227 [#zip-0227]_.
+- The term "Proportional Fee" in this document is to be interpreted as described in ZIP 317 [#zip-0317]_
+- Issuance Fee: is the specific fixed fee that has to be paid to the network for every given issuance action
+
+Abstract
+========
+
+This ZIP aims at defining explicitly the fee mechanism associated with the Zcash Shielded Assets protocol (ZSA) as described in ZIP 226 [#zip-0226]_ and ZIP 227 [#zip-0227]_. We will be basing this ZIP as an extension of the latest fee-related ZIP, ZIP 317 [#zip-0317]_.
+
+Motivation
+==========
+
+As the Zcash blockchain moves towards the Proportional Transfer Fee Mechanism proposed ZIP 317 [#zip-0317]_, transactions that use the ZSA protocol must adapt to this mechanism and include fees consistent with ZIP 317 [#zip-0317]_. 
+
+In this ZIP we propose an adaptation of ZIP 317 [#zip-0317]_ as it applies to ZSA transfer transactions and explore ZSA-specific considerations such as fees for asset issuance.
+
+Specifically, the ZSA transfer and burn mechanism should follow the same fee mechanism in order to not discriminate between transfer bundle types. When it comes to Issuance of ZSA, however, we believe that there should be a preventative incentive that will stop users from flooding the chain with useless asset identifiers.
+
+There are two main factors that will affect the fee mechanism:
+
+- The transaction size, which may take a big part of the block
+- The computational power needed to verify and mine the transaction
```

Comment:
What about the storage size and lookup cost for the asset global state?

This might include the `AssetId`, `asset_desc`, and `finalized` flag for every asset, which could be large in the case of NFTs.

(We could say that this covered because the global state size is proportional to the transaction size, and the lookup cost is proportional to the verification cost.)

Reply:
Right, we need to track the finalized assets as part of the global state. 

So we need to consider having a bigger fee for `finalized` assets which will make NFT minting more expansive then fungible tokens.